### PR TITLE
Fix and add test for starting enterprise in SINGLE mode.

### PR DESCRIPTION
**Pull request by martinfurmanski**

### CODE REVIEW

File: enterprise/server-enterprise/src/test/java/org/neo4j/server/enterprise/EnterpriseBootstrapperTest.java [hunk 1]
```diff
@@ -37,21 +38,42 @@ protected Bootstrapper newBootstrapper()
         return new EnterpriseBootstrapper();
     }
 
-    @Override
-    protected String[] baseConfig()
+    @Test
+    public void shouldBeAbleToStartInSingleMode() throws Exception
```
Comment (jakewins) [inline enterprise/server-enterprise/src/test/java/org/neo4j/server/enterprise/EnterpriseBootstrapperTest.java:42]: Nice.

File: enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseServerConfigLoader.java [hunk 2]
```diff
@@ -36,7 +36,10 @@
     public Config loadConfig( File configFile, File legacyConfigFile, Log log, Pair<String,String>... configOverrides )
     {
         Config config = super.loadConfig( configFile, legacyConfigFile, log, configOverrides );
-        config.registerSettingsClasses( asList( HaSettings.class, ClusterSettings.class ) );
+        if ( config.get( EnterpriseServerSettings.mode ).equals( "HA" ) )
```
Comment (jakewins) [inline enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseServerConfigLoader.java:39]: I honestly don't think we should do this here - it means we've duplicated this in both the factory and here. Adding it as a line in `HighlyAvailableEditionModule` means we solve the problem for both places, and sets a good precedent for registering settings classes along with the modules they configure.

**Comment by jakewins** [comment]
Merging this for now, since this resolves outstanding build problems, and my comment is something that can be resolved separately.

File: enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseServerConfigLoader.java (hunk 2 above)
Comment (martinfurmanski) [inline enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseServerConfigLoader.java:39]: I believe you are right, gonna look at changing it today! Thanks!

Comment (martinfurmanski) [inline enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseServerConfigLoader.java:39]: I was not able to create a testable solution. Although I think you are right in principle, it is just too messy to clean it up and create something reasonable and testable. I will have to leave it as it is.

**Pull request merged**